tests/evaluator/msbg/test_audiogram.py

```diff
@@ -43,3 +43,63 @@ def test_standard_audiograms_are_correct_severity():
     assert AUDIOGRAM_MILD.severity == "MILD"
     assert AUDIOGRAM_MODERATE.severity == "MODERATE"
     assert AUDIOGRAM_MODERATE_SEVERE.severity == "SEVERE"
+
+
+def test_audiogram_init():
+    """test construction of audiogram"""
+    levels = np.array([45, 45, 35, 45, 60, 65])
+    frequencies = np.array([250, 500, 1000, 2000, 4000, 6000])
+    audiogram = Audiogram(levels=levels, frequencies=frequencies)
+    assert np.all(audiogram.levels == levels)
+    assert np.all(audiogram.frequencies == frequencies)
+    assert isinstance(audiogram.levels, np.ndarray)
+
+
+def test_audiogram_init_error():
+    """test that cannot be constructed with invalid number of arguments"""
+    # Need to disable pylint so that we can test the invalid constructor calls
+    with pytest.raises(TypeError):
+        Audiogram(None)  # pylint: disable=no-value-for-parameter
+    with pytest.raises(ValueError):
+        Audiogram([1])  # pylint: disable=no-value-for-parameter
+    with pytest.raises(TypeError):
+        Audiogram(3, 4, 5, 6)  # pylint: disable=too-many-function-args
+
+
+@pytest.mark.parametrize(
+    "levels, frequencies, expected",
+    [
+        ([45], [250], "NOTHING"),
+        ([15], [2000], "NOTHING"),
+        ([16], [2000], "MILD"),
+        ([35], [2000], "MILD"),
+        ([36], [2000], "MODERATE"),
+        ([56], [2000], "MODERATE"),
+        ([57], [2000], "SEVERE"),
+        ([90], [8001], "NOTHING"),
+        ([16, 16], [8000, 8001], "MILD"),
+    ],
+)
+def test_severity(levels, frequencies, expected):
+    """test severity calculation"""
+    audiogram = Audiogram(levels=np.array(levels), frequencies=np.array(frequencies))
+    assert audiogram.severity == expected
+
+
+@pytest.mark.parametrize(
+    "requested_frequencies, expected_levels",
+    [
+        (np.array([250, 1000.0, 6000.0]), np.array([1.0, 3.0, 6.0])),
+        (np.array([260.0, 1000.0, 8000.0]), np.array([1.05658353, 3.0, 6.0])),
+    ],
+)
+def test_resample(requested_frequencies, expected_levels):
+    """test that a subset of cfs can be chosen"""
+    levels = np.array([1, 2, 3, 4, 5, 6])
+    frequencies = np.array([250, 500, 1000, 2000, 4000, 6000])
+    audiogram = Audiogram(levels=levels, frequencies=frequencies)
+
+    # Include cfs that don't exist in the requested subset
+    subset_audiogram = audiogram.resample(requested_frequencies)
+    assert np.allclose(subset_audiogram.frequencies, requested_frequencies)
+    assert np.allclose(subset_audiogram.levels, expected_levels)
```

Comment:
I thought you prefer the `pytest.approx` approach to `np` functions for validate test.

Reply:
Yes. Good point. I'll fix this.